framework/src/bundle/Bundle.cpp

```diff
@@ -316,6 +316,21 @@ Bundle::TimeStamp Bundle::GetLastModified() const
   return d->barchive ? d->barchive->GetLastModified() : d->timeStamp;
 }
 
+void* Bundle::GetSymbol(void * handle, const std::string& symname) const
+{
+  if(!d || !handle || symname.empty()) {
+      throw std::invalid_argument("invalid bundle");
```

Comment:
this exception can be thrown if more than the bundle is invalid. I'd suggest making the exception message a little more descriptive (to aid debugging).